edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/validations/ShippingInstructionValidator.java

```diff
@@ -149,6 +155,28 @@ private void validateShipper(ValidationState<ShippingInstruction> state) {
     }
   }
 
+  private void validateManifestFilings(ValidationState<ShippingInstruction> state) {
+    List<AdvanceManifestFilingEBL> advanceManifestFilingSIs = state.getValue().getAdvanceManifestFilings();
+    List<ConsignmentItem> consignmentItems = state.getValue().getConsignmentItems();
+    for(ConsignmentItem ci : consignmentItems ) {
+      List<AdvanceManifestFiling> advanceManifestFilings = ci.getShipment().getAdvanceManifestFilings();
+      List<AdvanceManifestFiling> matchedManifestFilings = advanceManifestFilings.stream()
+        .filter(two -> advanceManifestFilingSIs.stream()
+          .anyMatch(one -> one.getManifestTypeCode().equals(two.getManifestTypeCode())
+            && two.getCountryCode().equals(one.getCountryCode())))
+        .toList();
+      if (matchedManifestFilings.size() != advanceManifestFilingSIs.size()) {
+        state.getContext().buildConstraintViolationWithTemplate(
+            "This SI advance manifest filings mismatch with Shipment advance manifest filings. Make sure they are same" +
```

Comment:
Consider having the error message say that the client should correct the `carrierBookingReference`s to solve this error.

Reply:
"message": "Correct the Carrier Booking References [CRULPNUT7LFSJYQ9WLQWHLYUWQND8H2KTBV] to match the Shipping instructions"

Reply:
Which is the source of truth for Manifest filings ? is it the shipment or Shipping instruction?  If the provided Shipping instruction manifest codes doesnt match with shipment, Then I can provide " Manifest codes in the shipping instruction doesnt match with booking. Correct them in the Shipping instruction" Or If the Manifest Codes in the Shipping instruction are the source of truth, then I provide the message as "  Correct the manifest codes in Carrier Booking References [CRULPNUT7LFSJYQ9WLQWHLYUWQND8H2KTBV] to match the Shipping instructions". Its confusing to me..

Reply:
There are two things that need to be aligned at the same time:

1. An SI can link to multiple distinct Shipments (up to one per consignment item).  When doing so, **all** `Shipment`s must have a compatible advanced manifest filings.  This is one validation that need a clear error message if this property is not respected.

2. The SI must contain an advanced manifest filing that is a 1:1 match to the Shipments (and per one, all Shipments must be aligned, so if it matches one, it would match all).


Reading the code,  I think the current implementation is basically 100% doing 2) and it would as a side-effect happen to also detect 1) as a root cause.  However, my issue here is that the validation message should be different depending on whether the root cause is 1) or 2).

With 1), I need to change the carrier booking references to resolve the issue (no matter how I change the advanced manifest filings, the input will not work because the shipments are not aligned). For 2) I need to change the advanced manifest filings (add a missing one or remove one that should not be there).

I feel the validation messages should at least be able to reflect these two root causes of the problem.

Reply:
So basically, we need two validations

1) to make sure all shipment manifest filings are same.
2) the SI matches to the shipments manifest filings.

Reply:
Yes, and:

 * each validation will have its own validation message
 * if the 1) validation fails, skip the validation of 2) as it cannot succeed anyway. The error message from 2) would be confusing at best.

Reply:
You are right.. if first check fails then even second check fails as well.